cms/models.py

```diff
@@ -169,6 +170,15 @@ class ProgramIndexPage(CourseObjectIndexPage):
 
     def get_child_by_readable_id(self, readable_id):
         """Fetch a child page by the related Program's readable_id value"""
+        try:
+            queryset = ProgramRun.objects.annotate(
+                program_run=Concat("program__readable_id", models.Value("+"), "run_tag")
+            )
+            program_run = queryset.get(program_run=readable_id)
+            readable_id = program_run.program.readable_id
+        except ProgramRun.DoesNotExist:
+            pass
+
```

Comment:
What happens if we have an incorrect readable id coming in from the URL?

Reply:
doesn't matter it will proceed the way it was dealt before. 
It will give `404`.

<img width="1438" alt="Screenshot 2020-04-16 at 18 51 53" src="https://user-images.githubusercontent.com/4043989/79464412-75f5d300-8013-11ea-8249-05135406c490.png">

If it does not cover your question then can you please elaborate on that?